src/v/bytes/ioarray.cc

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "bytes/ioarray.h"
+
+#include "base/vassert.h"
+#include "bytes/iobuf_parser.h"
+
+#include <seastar/core/align.hh>
+#include <seastar/core/byteorder.hh>
+
+#include <sys/uio.h>
+
+#include <cstring>
+#include <iterator>
+
+ioarray::ioarray(size_t size)
+  : ioarray(uninitialized_t{}, size) {
+    for (size_t offset = 0, i = 0; offset < size;
+         offset += max_chunk_size, ++i) {
+        _buffers[i] = ss::temporary_buffer<char>(
+          std::min(max_chunk_size, size - offset));
+    }
+}
+
+ioarray::ioarray(ioarray::uninitialized_t, size_t size)
+  : _offset(0)
+  , _size(size)
+  , _buffers((size + max_chunk_size - 1) / max_chunk_size) {}
+
+ioarray& ioarray::operator=(ioarray&& o) noexcept {
+    if (this != &o) {
+        this->~ioarray();
+        new (this) ioarray(std::move(o));
+    }
+    return *this;
+}
+
+// NOLINTNEXTLINE(*swappable-parameters*)
+ioarray ioarray::aligned(size_t alignment, size_t size) {
+    if (size == 0) {
+        return ioarray{0};
+    }
+    dassert(
+      (alignment & (alignment - 1)) == 0,
+      "alignment {} must be a power of two",
+      alignment);
+    dassert(
+      size % alignment == 0,
+      "size {} must be a multiple of alignment {}",
+      size,
+      alignment);
+    ioarray c{size};
+    for (size_t offset = 0, i = 0; offset < size;
+         offset += max_chunk_size, ++i) {
+        size_t len = std::min(max_chunk_size, size - offset);
+        auto tmpbuf = ss::temporary_buffer<char>::aligned(
+          alignment, ss::align_up(len, alignment));
+        tmpbuf.trim(len); // trim to actual size
+        c._buffers[i] = std::move(tmpbuf);
+    }
+    return c;
+}
+
+ioarray::string_view::operator ss::sstring() const {
```

Comment:
Maybe a silly question, but why does the LSM tree need to operate on strings? Why couldn't it operate on iobufs, for example?